pkg/otlp/logs/logs_translator_test.go

```diff
@@ -410,6 +410,113 @@ func TestTransform(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "Nestings",
+			args: args{
+				lr: func() plog.LogRecord {
+					l := plog.NewLogRecord()
+					l.Attributes().FromRaw(
+						map[string]any{
+							"root": map[string]any{
+								"nest1": map[string]any{
+									"nest2": "val",
+								},
+								"nest12": map[string]any{
+									"nest22": map[string]any{
+										"nest3": "val2",
+									},
+								},
+								"nest13": "val3",
+							},
+						},
+					)
+					return l
+				}(),
+				res: func() pcommon.Resource {
+					r := pcommon.NewResource()
+					return r
+				}(),
+			},
+			want: datadogV2.HTTPLogItem{
+				Ddtags:  datadog.PtrString(""),
+				Message: *datadog.PtrString(""),
+				AdditionalProperties: map[string]string{
+					"root.nest1.nest2":         "val",
+					"root.nest12.nest22.nest3": "val2",
+					"root.nest13":              "val3",
+					"status":                   "",
+				},
+			},
+		},
+		{
+			name: "Nil Map",
+			args: args{
+				lr: func() plog.LogRecord {
+					l := plog.NewLogRecord()
+					l.Attributes().FromRaw(nil)
+					return l
+				}(),
+				res: func() pcommon.Resource {
+					r := pcommon.NewResource()
+					return r
+				}(),
+			},
+			want: datadogV2.HTTPLogItem{
+				Ddtags:  datadog.PtrString(""),
+				Message: *datadog.PtrString(""),
+				AdditionalProperties: map[string]string{
+					"status": "",
+				},
+			},
+		},
+		{
+			name: "Too many nestings",
```

Comment:
It might be good to have a test that combines the attributes of "Too many nestings" with "Nestings" to verify that it still flattens properly